data_steward/tools/run_deid.py

```diff
@@ -211,9 +248,69 @@ def parse_args(raw_args=None):
                         required=False,
                         help='Log to the console as well as to a file.')
     parser.add_argument('--version', action='version', version='deid-02')
+    parser.add_argument(
+        '-ma',
+        '--age_limit',
+        dest='age_limit',
+        action='store',
+        required=True,
+        help=
+        'Determines the max_age of a participant that is to be allowed in the cdr.'
+    )
     return parser.parse_args(raw_args)
 
 
+def copy_deid_map_table(pid_rid_mapping_table, project_id, input_dataset_id,
+                        output_dataset_id, age_limit, client):
+    """
+    Copies research_ids for participants who's age is below max_age limit from pipeline_tables.pid_rid_mapping table
+     to _deid_map table.
+
+    :param pid_rid_mapping_table: Fully Qualified(fq) pid_rid_mapping table name
+    :param project_id: Project identifier 
+    :param input_dataset_id: Name of the dataset where pid_rid_mapping table is stored
+    :param output_dataset_id: Name of the dataset where _deid_map dataset needs to be created.
+    :param age_limit: Allowed Max_age of a participant
+    :param client: Bigquery client
+    :return: None
+    """
+    job_config = bq.bigquery.QueryJobConfig(destination=pid_rid_mapping_table)
+    q = COPY_PID_RID_QUERY.render(project=project_id,
+                                  dataset=input_dataset_id,
+                                  output_dataset=output_dataset_id,
+                                  max_age=age_limit,
+                                  pid_rid_table=PID_RID_MAPPING_TABLE)
+
+    query_job = client.query(q, job_config=job_config)
+    query_job.result()
+    if query_job.exception():
+        logging.error(f"The _deid_map table did not generate successfully")
+
+
+def load_deid_map_table(deid_map_dataset_name, age_limit):
+
+    # Create _pid_rid_mapping table
+    project_id = app_identity.get_application_id()
+    client = bq.get_client(project_id)
+    deid_map_mapping_table = f'{project_id}.{deid_map_dataset_name}.{DEID_MAP_TABLE}'
+    bq.create_tables(client,
+                     project_id, [deid_map_mapping_table],
+                     exists_ok=False)
+    # Copy pid_rid_mapping records to _deid_map table
+    if bq_utils.table_exists(PID_RID_MAPPING_TABLE,
+                             dataset_id=PIPELINE_TABLES_DATASET):
```

Comment:
Did you consider using a DDL statement to create the table?  What are the advantages/disadvantages the explicitly creating the table and checking for its existence instead of using a DDL statement?  It seems like a `Create Table If Not Exists` [DDL statement](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#creating_a_table_only_if_the_table_doesnt_exist) would achieve the same results with fewer API calls.  If you did use a DDL statement, I think you would want to add a `try/except/else` to catch and log when the [table already exists.](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#creating_a_new_table)

I think I may be missing a design decision.  I'm definitely interested in your thoughts.

Reply:
You are right I should have used DDL in the first place. It didn't come into my mind while I was adding an API call.